### DISCUSSION TITLE
Build code-server in Raspberry Pi

### DISCUSSION THREAD

**Discussion by zhangrongxiang** [coder/code-server#3932] [Q&A]
I tried to install code-server in Raspberry PI3B+ by npm.  
I used the code:
`npm install -g code-server`   

But it reported:
```npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! argon2@0.28.2 install: node-pre-gyp install --fallback-to-build
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the argon2@0.28.2 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-08-09T01_28_29_522Z-debug.log
```

**Comment by lg-nineoneninethreenine** [comment]
Try using curl. Curl could work.
The line:

```bash
curl -fsSL https://code-server.dev/install.sh | sh
```

**Reply by lg-nineoneninethreenine** [reply]
> I tried to install code-server in Raspberry PI3B+ by npm.  
> I used the code:
> `npm install -g code-server`   
> 
> But it reported:
> ```npm ERR! code ELIFECYCLE
> npm ERR! errno 1
> npm ERR! argon2@0.28.2 install: node-pre-gyp install --fallback-to-build
> npm ERR! Exit status 1
> npm ERR!
> npm ERR! Failed at the argon2@0.28.2 install script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /root/.npm/_logs/2021-08-09T01_28_29_522Z-debug.log
> ```

it works best via curl.

**Comment by jsjoeio** [comment]
Hopefully the next release will make it easier to use code-server on Pi: https://github.com/cdr/code-server/pull/3892

**Reply by bpmct** [reply]
wohoo!
